lib/model/model.go

```diff
@@ -565,16 +565,16 @@ func (m *model) restartFolder(from, to config.FolderConfiguration, cacheIgnoredF
 }
 
 func (m *model) newFolder(cfg config.FolderConfiguration, cacheIgnoredFiles bool) error {
-	// Creating the fileset can take a long time (metadata calculation) so
-	// we do it outside of the lock.
+	m.mut.Lock()
+	defer m.mut.Unlock()
+
+	// Creating the fileset can take a long time (metadata calculation), but
+	// nevertheless should happen inside the lock.
```

Comment:
Maybe worth referencing the other place here (and potentially vice-versa), just so if we decide to change it once again, we at least increase the chance that we consider both places :) 

```suggestion
	// Creating the fileset can take a long time (metadata calculation), but
	// nevertheless should happen inside the lock (same as when restarting
	// a folder).
```